### ISSUE TITLE
bug: pointmap confidence is unbounded, so train loss goes negative and `val/loss` is unusable for model selection

### ISSUE THREAD

**Issue by engichang1467** [engichang1467/Open-Rig3R#53]
Found while diagnosing an apparent overfit in a 10-epoch Waymo run. The overfit is real,
but `val/loss` reports it for the wrong reason and points at the wrong epoch. The
confidence-weighted pointmap term runs away, and it is the only term that gets worse.

Run: `runs/wandb/offline-run-20260812_221242-43p857tq`, `configs/train_waymo.yaml` with
`epochs: 10`, `/data/waymo_mini` (5 train segments, 5 validation segments, 985 train
samples, 247 steps/epoch), 724 s on one GPU.

## Problem

`models/losses.py:59` implements Eq. 3's confidence weighting:

```python
term = conf * error - self.alpha * conf.log()
```

`models/heads/pointmap_head.py:126` produces `conf`:

```python
confidence = 1.0 + F.softplus(confidence)
```

`softplus` is unbounded above, so `conf` is unbounded above, so `-alpha * log(conf)` is
unbounded below. Minimising over `conf` for a fixed `error` gives `conf = alpha / error`,
and the term at that optimum is `alpha * (1 - log(alpha) + log(error))`, which goes to
`-inf` as `error` goes to zero. There is no floor.

This is faithful to DUSt3R and to Eq. 3 — it is not a transcription bug. DUSt3R survives
it because its training error never approaches zero across millions of diverse pairs.
On 5 segments of `waymo_mini` the training error does approach zero, and the term
collapses through it.

## Evidence

Per-epoch mean of `train/batch_pointmap` over all 247 batches, and the fraction of
batches where that term is negative:

| epoch | mean | min | negative batches |
|---|---|---|---|
| 1 | 0.4405 | 0.2158 | 0% |
| 2 | 0.1971 | 0.1007 | 0% |
| 3 | 0.1323 | 0.0512 | 0% |
| 4 | 0.0873 | 0.0060 | 0% |
| 5 | 0.0430 | −0.0328 | 7% |
| 6 | 0.0163 | −0.0672 | 34% |
| 7 | **−0.0054** | −0.0820 | 57% |
| 8 | −0.0246 | −0.1332 | 72% |
| 9 | −0.0450 | −0.1428 | 91% |
| 10 | −0.0623 | −0.1463 | **95%** |

The epoch mean crosses zero at epoch 7. By epoch 10, `train/batch_total` itself is
negative on 19% of batches (min −0.0712), i.e. the whole multi-task objective — pointmap
plus both raymaps — is being driven below zero by the regulariser alone. `train/loss` is
still reported as a positive 0.0444 because the positive raymap terms average over it.

A loss that can be reduced without bound by inflating one scalar is not measuring
reconstruction quality any more. The optimiser has found that.

## The consequence: `val/loss` blames the wrong thing

Validation terms from the same run:

| epoch | `val/loss` | pointmap | pose_dir | rig_dir | pose_center | rig_center | pose_deg | rig_deg |
|---|---|---|---|---|---|---|---|---|
| 1 | 2.336 | 0.783 | 0.716 | 0.727 | 0.036 | 0.074 | 46.17 | 46.87 |
| 3 | **2.156** | 0.780 | 0.648 | 0.652 | 0.024 | 0.052 | 41.24 | 41.58 |
| 5 | 2.321 | 0.988 | 0.629 | 0.633 | 0.022 | 0.049 | 41.29 | 41.60 |
| 7 | 2.337 | 1.079 | 0.591 | 0.593 | 0.024 | 0.050 | 39.32 | 39.45 |
| 10 | 2.681 | **1.436** | 0.585 | 0.586 | 0.021 | 0.052 | 39.26 | 39.34 |

Epoch 3 to epoch 10, by term:

```
pointmap      +0.656
pose_raymap   -0.066
rig_raymap    -0.066
              ------
              +0.524     (val/loss actually moved +0.525)
```

The pointmap term accounts for the entire regression and then some. Every other
validation term improves monotonically, and so do the two loss-independent geometric
metrics: `pose_deg` 46.17 to 38.89, `rig_deg` 46.87 to 38.92, both still falling at
epoch 9.

The mechanism on the validation side is the same inflated `conf`. It is predicted
per-pixel from the image, so the "be confident" behaviour transfers to held-out
segments, where `error` does *not* go to zero — and `conf * error` grows accordingly.

So `val/loss` has a minimum at epoch 3 while held-out geometry is best at epoch 9.
Anything that selects on `val/loss` picks a worse model. The checkpoint block at
`scripts/train.py:412-424` is currently commented out, but it already records
`metadata={"epoch": epoch + 1, "val_loss": avg_val_loss}` — so if it is re-enabled and
a "best" checkpoint is chosen by that field, it will select epoch 3.

## Suggested fix

**1. Log the raw error and the mean confidence separately.** This is the gap that made
the diagnosis take a run-file parse. `loss_dict['pointmap']` bundles `conf * error` and
`-alpha * log(conf)` into one number, so there is no way to tell from the logs whether
held-out reconstruction is degrading or only `conf` is inflating — and those call for
opposite responses. In `MultiTaskLoss.forward`, alongside `loss_dict['pointmap']` at
`models/losses.py:66`:

```python
loss_dict['pointmap_err'] = self._reduce_masked(error, gts.get('pointmap_conf'))
loss_dict['pointmap_conf'] = self._reduce_masked(conf, gts.get('pointmap_conf'))
```

Both flow into `train/batch_*` and `val/*` with no other change, since `train.py` already
logs every key in `loss_dict`. Cheap, and it makes every future run readable.

Do this first regardless of what is decided about 2 and 3 — it is also what tells us
whether `val/pointmap` rising at all reflects worse geometry.

**2. Stop selecting on `val/loss`.** It is not comparable across epochs once one term is
unbounded. `val/pose_deg` and `val/rig_deg` are already logged, already
loss-independent by construction (see the note at the top of `utils/metrics.py`), and
already say something a reader can interpret. Whatever early stopping or best-checkpoint
logic lands should key on those, not on the total.

**3. Bound the confidence — decide deliberately.** Three options, in order of how much
they deviate from the paper:

- Leave it, and treat the runaway as a symptom of `waymo_mini` being too small.
  Defensible: the term is faithful to Eq. 3, and on a full-size dataset it may never
  trigger. Costs nothing now, but every `waymo_mini` run stays unreadable on this term.
- Clamp: `conf.clamp(max=C)` for some `C`, which floors the term at
  `-alpha * log(C)`. One line, keeps the shape of the objective everywhere below the
  clamp. `C` is a new hyperparameter with no value from the paper — same situation
  `alpha` and `beta` are already in.
- Drop the confidence weighting on small datasets by setting the head to a constant
  `conf = 1`, which reduces the term to plain `error`. Simplest to reason about, loses
  the ability to downweight occluded regions.

No recommendation between these without 1 landing first — the raw-error curve decides
whether held-out reconstruction is actually degrading or whether `conf` is the entire
story.

## Not included

Whether `/data/waymo_mini` is too small is a separate question and is not resolved here.
5 training segments at 985 stride-1 sliding windows is roughly 5 independent scenes of
20 s, and stride-1 windows at 10 Hz share a frame with their neighbour — the training
error reaching zero is unsurprising. But the confidence runaway is what turns that into
an unbounded loss, and it would be worth fixing on a larger dataset too. Filing the data
question separately if it turns out to matter.

## Related finding: `scheduler.T_max` does not track `epochs`

Found while reading the same run. `configs/train_waymo.yaml` sets `scheduler.T_max: 50`
independently of `epochs`, so a 10-epoch run traverses one fifth of the cosine period:
`lr` moved 9.99e-5 to 9.05e-5 across the whole run, a 9% decay. Effectively a constant
learning rate.

Harmless when `epochs: 50`, silently wrong otherwise, and nothing connects the two keys.
Either default `T_max` to `epochs` in `scripts/train.py`, or validate that they match and
fail loudly. Separable from the above — split it out if preferred.

## Reproducing

```bash
source .venv/bin/activate
python scripts/train.py --config configs/train_waymo.yaml   # with epochs: 10
```

The run above is offline and its `files/` contains only `requirements.txt` — no
`config.yaml`, no `wandb-summary.json`, so `run.finish()` did not flush. History and
summary are still readable directly from `run-43p857tq.wandb` via
`wandb.sdk.internal.datastore`. Worth checking separately why the finish did not
complete, since it means the exact config of this run is not recoverable from the run
directory.
